addons/website_slides/models/slide_slide.py

```diff
@@ -266,6 +277,24 @@ def _compute_category_id(self):
                 elif slide.category_id != current_category:
                     slide.category_id = current_category.id
 
+    @api.depends('slide_type', 'question_ids', 'channel_id.is_member')
+    @api.depends_context('uid')
+    def _compute_mark_complete_actions(self):
+        """Determine if the slide can be marked as (un)completed.
+
+        We can't mark a slide with questions as completed manually because we need to
+        complete the quiz first. But we can mark as uncompleted a slide with questions,
+        and the answers will be reset, the karma removed, etc (see mark_uncompleted).
+        """
+        for slide in self:
+            slide.can_self_mark_uncompleted = slide.website_published and slide.channel_id.is_member
+            slide.can_self_mark_completed = (
+                slide.website_published
+                and slide.channel_id.is_member
+                and slide.slide_type != 'quiz'
```

Comment:
Even if it does not matter too much for this case, you should be using the slide_category here.
(To adapt in depends as well).

The "slide_type" field is now mostly used for display and is a lot more precise that slide_category.
For quiz it's the same so it should still work, but I would prefer to do the check on slide_category as it is the one the end user defines on the model.